[Waiting Room] document third-party cookie limitations (#13525)
cloudflare/cloudflare-docs: content/waiting-room/additional-options/embed-waiting-room-in-iframe.md

```

{{</details>}}

## Limitations

> Major browsers have introduced restrictions on third-party cookies, the very kind of cookie

Reviewer: ```suggestion
Major web browsers have introduced restrictions on third-party cookies, which happen to be the same type of cookies used by waiting rooms within iframes. Waiting Room uses [Cookies Having Independent Partitioned State (CHIPS)](https://developer.mozilla.org/en-US/docs/Web/Privacy/Privacy_sandbox/Partitioned_cookies) to work around these restrictions, but there are some drawbacks:
```

> used by waiting rooms in iframes. Waiting Room uses [Cookies Having Independent Partitioned State (CHIPS)](https://developer.mozilla.org/en-US/docs/Web/Privacy/Privacy_sandbox/Partitioned_cookies)

Reviewer: ```suggestion
```

> to work around these restrictions, but there are some drawbacks:

Reviewer: ```suggestion
```

>  - A user viewing the waiting room both within an iframe and outside the iframe will be

Reviewer: ```suggestion
 - A user viewing the waiting room both within an iframe and outside the iframe will be treated as two separate users, with each instance potentially exiting the queue at different times and counting separately in analytics.
```

>    treated as two separate users, with each instance potentially exiting the queue at

Reviewer: ```suggestion
```

>    different times and counting separately in analytics.

Reviewer: ```suggestion
```

>  - For a waiting room to be embedded in an iframe, both the embedded page and the embedding

Reviewer: ```suggestion
 - For a waiting room to be embedded in an iframe, both the embedded page and the embedding page must be accessed over HTTPS.
```

>    page must be accessed over HTTPS.

Reviewer: ```suggestion
```

>  - CHIPS is not supported on Safari or Safari-derived browsers (e.g. Orion and most iOS browsers)

Reviewer: ```suggestion
 - CHIPS is not supported on Safari or Safari-derived browsers, like Orion and most iOS browsers, unless they have third-party cookie blocking disabled in their settings. These users will be stuck at the end of the queue, unable to progress until the queue is empty, and may count multiple times in analytics.
```

>    unless they have third-party cookie blocking disabled in their settings. These users will be

Reviewer: ```suggestion
```

>    stuck at the end of the queue, unable to progress until the queue is empty, and may count

Reviewer: ```suggestion
```

>    multiple times in analytics.

Reviewer: ```suggestion
```

> In general, if there is an issue setting and retrieving the waiting room cookie, you should

Reviewer: ```suggestion
In general, if there is an issue setting and retrieving the waiting room cookie, you should expect users to be stuck at the end of the queue, and counting as multiple users in analytics.
```

> expect users to be stuck at the end of the queue, and counting as multiple users in analytics.

Reviewer: ```suggestion
```

> These limitations may not apply if the embedded page and embedding page share a common domain

Reviewer: ```suggestion
These limitations may not apply if the embedded page and embedding page share a common domain name. For example, a page at `example.com` embedding a waiting room at `shop.example.com` may be considered first party by browsers, and not subject to third-party cookie restrictions.
```

> name. For example, a page at `example.com` embedding a waiting room at `shop.example.com` may

Reviewer: ```suggestion
```

> be considered first party by browsers, and not subject to third-party cookie restrictions.

Reviewer: ```suggestion
```
